### PR TITLE
Generalize to a pyenv setup script

### PR DESCRIPTION
**Pull request by JayH5**
- The PyPy-specific parts are small

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -24,23 +26,24 @@ There are a few install options that can be set via environment variables:
...
-        wget https://github.com/praekeltfoundation/travis-pypy/releases/download/0.1.0/setup-pypy.sh
-        source setup-pypy.sh
+      if [[ -n "$PYENV_VERSION" ]]; then
+        git clone https://github.com/praekeltfoundation/travis-pyenv.git ~/travis-pyenv
+        source ~/travis-pyenv/setup-pyenv.sh
       fi
 
 script:
-  - py.test my_project
+  - pytest my_project
```
Comment (jerith) [inline README.md]: Why did this change?

**Review by jerith — COMMENTED** [review]
Do we want to continue to support the old `PYPY_VERSION` envvar? If we don't, will existing things break?

File: README.md (hunk 1 above)
Comment (JayH5) [reply README.md]: Not strictly necessary. See the pytest changelog for 3.0.0: http://docs.pytest.org/en/latest/changelog.html#id18

_"Introduce pytest command as recommended entry point. Note that py.test still works and is not scheduled for removal."_

**Comment by JayH5** [comment]
This is not quite ready yet :p

I'm hoping most existing things are using the tagged release of this so `PYPY_VERSION` will continue to work.. but will check.

File: README.md (hunk 1 above)
Comment (jerith) [reply README.md]: I'd prefer to stick with `py.test` which has existed for longer than a couple of months. :-)

**Comment by jerith** [comment]
I know it's not ready yet. I just thought that some early comments might be useful. :-)

File: README.md [hunk 2]
```diff
@@ -24,19 +26,20 @@ There are a few install options that can be set via environment variables:
 language: python
 matrix:
   include:
-    - python: "2.7"
-    - python: "pypy"
-      env: PYPY_VERSION="5.3.1"
+    - python: '2.7'
+    - python: '3.5'
+    - python: pypy
+      env: PYENV_VERSION="pypy-5.4.1" PYENV_VERSION_STRING="PyPy 5.4.1"
```
Comment (jerith) [inline README.md]: Why not switch these `"` to `'` as well?

File: README.md [hunk 3]
```diff
@@ -24,19 +26,20 @@ There are a few install options that can be set via environment variables:
...
+    - ~/.pyenv_cache
 
 before_install:
   - |
-      if [[ -n "$PYPY_VERSION" ]]; then
-        wget https://github.com/praekeltfoundation/travis-pypy/releases/download/0.1.0/setup-pypy.sh
-        source setup-pypy.sh
+      if [[ -n "$PYENV_VERSION" ]]; then
+        git clone https://github.com/praekeltfoundation/travis-pyenv.git ~/travis-pyenv
+        source ~/travis-pyenv/setup-pyenv.sh
```
Comment (jerith) [inline README.md:44]: Why do we clone the repo instead of downloading a released script now?

Comment (jerith) [reply README.md:44]: I think a versioned download is more sensible than a git clone that could have anything in it.

Comment (JayH5) [reply README.md:44]: Only reason is that there isn't a released version yet and I didn't want to include a broken command.

File: README.md (hunk 2 above)
Comment (JayH5) [reply README.md]: Just me failing to keep track of all the tiny changes 😅 thanks

**Review by jerith — APPROVED** [review]
:+1:

**Pull request merged**